src/reanimated2/hook/useAnimatedStyle.ts

```diff
@@ -442,27 +442,29 @@ For more, see the docs: \`https://docs.swmansion.com/react-native-reanimated/doc
   }
   adaptersHash && dependencies.push(adaptersHash);
 
-  if (!initRef.current) {
+  if (!propUpdaterData.current) {
     const initialStyle = initialUpdaterRun(updater);
-    validateAnimatedStyles(initialStyle);
-    initRef.current = {
+    if (__DEV__) {
+      validateAnimatedStyles(initialStyle);
+    }
+    propUpdaterData.current = {
       initial: {
         value: initialStyle,
-        updater: updater,
+        updater,
       },
-      remoteState: makeRemote<AnimatedState>({
+      remoteState: {
```

Comment:
Why is it safe to remove `makeRemote` here?

Reply:
It's now in PR's description.

Reply:
@bartlomiejbloniarz Here's another call to `makeRemote` that was removed in this PR. Can you please confirm if it doesn't break anything? Maybe we should introduce a fix similar to #5586 here? We can also rename `remoteState` to `uiState`.